File: redome/src/main/java/br/org/cancer/modred/model/formulario/Pergunta.java
package br.org.cancer.modred.model.formulario;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonView;

import br.org.cancer.modred.controller.view.FormularioView;
import br.org.cancer.modred.model.domain.TiposPergunta;

/**
 * Classe que representa a pergunta de um formulário.
 * 
 * @author bruno.sousa
 *
 */
public class Pergunta implements Serializable {
	
	private static final long serialVersionUID = 6490501007969927857L;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String marcador;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String id;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String descricao;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String tipo;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private Boolean possuiDependencia = false;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private Long tamanho = 0L;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String resposta;
	
	private String valorDefault = "";
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private List<OpcaoPergunta> opcoes;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private List<PerguntaDependente> dependentes;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private FormatoNumerico formatoNumerico;
	
	private CondicaoPergunta condicao;
	
	private Boolean removerPergunta = false;

	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private Boolean comValidacao = true;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private Boolean comErro = false;
	
	@JsonView(value = { FormularioView.Formulario.class, FormularioView.FormularioRespondido.class})
	private String alinhamento = "V";
	
	/**
	 * Contrutor padrão.
	 */
	public Pergunta() {
	}
	
	/**
	 * @return the marcador
	 */
	public String getMarcador() {
		return marcador;
	}

	/**
	 * @param marcador the marcador to set
	 */
	public void setMarcador(String marcador) {
		this.marcador = marcador;
	}

	/**
	 * Identificador da pergunta <b>Token</b>.
	 * 
	 * @return the id
	 */
	public String getId() {
		return id;
	}

	
	/**
	 * Identificador da pergunta <b>Token</b>.
	 * 
	 * @param id the id to set
	 */
	public void setId(String id) {
		this.id = id;
	}

	
	/**
	 * Descrição da pergunta.
	 * 
	 * @return the descricao
	 */
	public String getDescricao() {
		return descricao;
	}

	
	/**
	 * Descrição da pergunta.
	 * 
	 * @param descricao the descricao to set
	 */
	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}

	
	/**
	 * Tipo da Pergunta.
	 * 
	 * @return the tipo
	 */
	public TiposPergunta getTipo() {
		return TiposPergunta.valueOf(tipo);
	}

	
	/**
	 * Tipo da Pergunta.
	 * 
	 * @param tipo the tipo to set
	 */
	public void setTipo(TiposPergunta tipo) {
		if (tipo != null) {
			this.tipo = tipo.toString();
		}
	}

	
	/**
	 * Informa se a pergunta possui depedência de outra pergunta.
	 * 
	 * @return the possuiDependencia
	 */
	public Boolean getPossuiDependencia() {
		return possuiDependencia;
	}
	
	/**
	 * Informa se a pergunta possui depedência de outra pergunta.
	 * 
	 * @param possuiDependencia the possuiDependencia to set
	 */
	public void setPossuiDependencia(Boolean possuiDependencia) {
		this.possuiDependencia = possuiDependencia;
	}
	
	/**
	 * Informa a quantidade de caracteres para os tipos TEXT e TEXTAREA.
	 * 
	 * @return the tamanho
	 */
	public Long getTamanho() {
		return tamanho;
	}
	
	/**
	 * Informa a quantidade de caracteres para os tipos TEXT e TEXTAREA.
	 * 
	 * @param tamanho the tamanho to set
	 */
	public void setTamanho(Long tamanho) {
		this.tamanho = tamanho;
	}
	
	/**
	 * Valor da resposta da pergunta.
	 * 
	 * @return the resposta
	 */
	public String getResposta() {
		return resposta;
	}
	
	/**
	 * Valor da resposta da pergunta.
	 * 
	 * @param resposta the resposta to set
	 */
	public void setResposta(String resposta) {
		this.resposta = resposta;
	}
	
	/**
	 * Valor default para a pergunta, podendo ser um valor qualquer ou um token.
	 * Caso exista valor nesta propriedade, será colocada na prorpriedade resposta.
	 * 
	 * @return the valorDefault
	 */
	public String getValorDefault() {
		return valorDefault;
	}
	
	/**
	 * Valor default para a pergunta, podendo ser um valor qualquer ou um token.
	 * Caso exista valor nesta propriedade, será colocada na prorpriedade resposta.
	 * 
	 * @param valorDefault the valorDefault to set
	 */
	public void setValorDefault(String valorDefault) {
		this.valorDefault = valorDefault;
	}
	
	/**
	 * Lista de opções para os tipos RADIO, CHECK ou SELECT.
	 * 
	 * @return the opcoes
	 */
	public List<OpcaoPergunta> getOpcoes() {
		return opcoes;
	}
	
	/**
	 * Lista de opções para os tipos RADIO, CHECK ou SELECT.
	 * 
	 * @param opcoes the opcoes to set
	 */
	public void setOpcoes(List<OpcaoPergunta> opcoes) {
		this.opcoes = opcoes;
	}
	
	/**
	 * Lista de perguntas dependentes.
	 * 
	 * @return the dependentes
	 */
	public List<PerguntaDependente> getDependentes() {
		return dependentes;
	}
	
	/**
	 * Lista de perguntas dependentes.
	 * 
	 * @param dependentes the dependentes to set
	 */
	public void setDependentes(List<PerguntaDependente> dependentes) {
		this.dependentes = dependentes;
	}

	/**
	 * @return the condicao
	 */
	public CondicaoPergunta getCondicao() {
		return condicao;
	}

	/**
	 * @param condicao the condicao to set
	 */
	public void setCondicao(CondicaoPergunta condicao) {
		this.condicao = condicao;
	}

	/**
	 * @return the removerPergunta
	 */
	public Boolean getRemoverPergunta() {
		return removerPergunta;
	}

	/**
	 * @param removerPergunta the removerPergunta to set
	 */
	public void setRemoverPergunta(Boolean removerPergunta) {
		this.removerPergunta = removerPergunta;
	}

	/**
	 * @return the formatoNumerico
	 */
	public FormatoNumerico getFormatoNumerico() {
		return formatoNumerico;
	}

	/**
	 * @param formatoNumerico the formatoNumerico to set
	 */
	public void setFormatoNumerico(FormatoNumerico formatoNumerico) {
		this.formatoNumerico = formatoNumerico;
	}
	
	/**
	 * @return the comValidacao
	 */
	public Boolean getComValidacao() {
		return comValidacao;
	}

	/**
	 * @param comValidacao the comValidacao to set
	 */
	public void setComValidacao(Boolean comValidacao) {
		this.comValidacao = comValidacao;
	}

	/**
	 * @return the comErro
	 */
	public Boolean getComErro() {
		return comErro;
	}

	/**
	 * @param comErro the comErro to set
	 */
	public void setComErro(Boolean comErro) {
		this.comErro = comErro;
	}

	/**
	 * @return the alinhamento
	 */
	public String getAlinhamento() {
		return alinhamento;
	}

	/**
	 * @param alinhamento the alinhamento to set
	 */
	public void setAlinhamento(String alinhamento) {
		this.alinhamento = alinhamento;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ( ( dependentes == null ) ? 0 : dependentes.hashCode() );
		result = prime * result + ( ( descricao == null ) ? 0 : descricao.hashCode() );
		result = prime * result + ( ( id == null ) ? 0 : id.hashCode() );
		result = prime * result + ( ( opcoes == null ) ? 0 : opcoes.hashCode() );
		result = prime * result + ( ( possuiDependencia == null ) ? 0 : possuiDependencia.hashCode() );
		result = prime * result + ( ( resposta == null ) ? 0 : resposta.hashCode() );
		result = prime * result + ( ( tamanho == null ) ? 0 : tamanho.hashCode() );
		result = prime * result + ( ( tipo == null ) ? 0 : tipo.hashCode() );
		result = prime * result + ( ( valorDefault == null ) ? 0 : valorDefault.hashCode() );
		return result;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#equals(java.lang.Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (!( obj instanceof Pergunta )) {
			return false;
		}
		Pergunta other = (Pergunta) obj;
		if (dependentes == null) {
			if (other.dependentes != null) {
				return false;
			}
		}
		else
			if (!dependentes.equals(other.dependentes)) {
				return false;
			}
		if (descricao == null) {
			if (other.descricao != null) {
				return false;
			}
		}
		else
			if (!descricao.equals(other.descricao)) {
				return false;
			}
		if (id == null) {
			if (other.id != null) {
				return false;
			}
		}
		else
			if (!id.equals(other.id)) {
				return false;
			}
		if (opcoes == null) {
			if (other.opcoes != null) {
				return false;
			}
		}
		else
			if (!opcoes.equals(other.opcoes)) {
				return false;
			}
		if (possuiDependencia == null) {
			if (other.possuiDependencia != null) {
				return false;
			}
		}
		else
			if (!possuiDependencia.equals(other.possuiDependencia)) {
				return false;
			}
		if (resposta == null) {
			if (other.resposta != null) {
				return false;
			}
		}
		else
			if (!resposta.equals(other.resposta)) {
				return false;
			}
		if (tamanho == null) {
			if (other.tamanho != null) {
				return false;
			}
		}
		else
			if (!tamanho.equals(other.tamanho)) {
				return false;
			}
		if (tipo == null) {
			if (other.tipo != null) {
				return false;
			}
		}
		else
			if (!tipo.equals(other.tipo)) {
				return false;
			}
		if (valorDefault == null) {
			if (other.valorDefault != null) {
				return false;
			}
		}
		else
			if (!valorDefault.equals(other.valorDefault)) {
				return false;
			}
		return true;
	}

}

File: redome-front/src/app/shared/classes/pergunta.ts
import { ConvertUtil } from 'app/shared/util/convert.util';
import { BaseEntidade } from '../base.entidade';
import { FormatoNumerico } from './formato.numerico';
import { OpcaoPergunta } from './opcao.pergunta';
import { PerguntaDependente } from './pergunta.dependente';

/**
 * Classe que representa uma pergunta de um formulário.
 * 
 * @export
 * @class Pergunta
 */
export class Pergunta extends BaseEntidade {

	private _marcador: string;
	private _id: string;
	private _descricao: string;
	private _tipo: String;
	private _possuiDependencia: boolean;
	private _tamanho: number;
	private _resposta: string;
	private _opcoes: OpcaoPergunta[] = [];
	private _dependentes: PerguntaDependente[] = [];
	private _formatoNumerico: FormatoNumerico;
	private _secao: number;
	private _comValidacao: boolean;
	private _comErro: boolean;
	private _alinhamento: string;

 	/**
	 * Getter marcardor
	 * @return {string}
	 */
	public get marcador(): string {
		return this._marcador;
	}

	/**
	 * Setter marcardor
	 * @param {string} value
	 */
	public set marcador(value: string) {
		this._marcador = value;
	}

	/**
	 * Identificador da pergunta
	 * 
	 * @type {string}
	 * @memberOf Pergunta
	 */
	public get id(): string {
		return this._id;
	}

	/**
	 * Identificador da Pergunta
	 * 
	 * 
	 * @memberOf Pergunta
	 */
	public set id(value: string) {
		this._id = value;
	}

	/**
	 * Descrição da pergunta
	 * 
	 * @type {string}
	 * @memberOf Pergunta
	 */
	public get descricao(): string {
		return this._descricao;
	}

	/**
	 * Descrição da pergunta
	 * 
	 * 
	 * @memberOf Pergunta
	 */
	public set descricao(value: string) {
		this._descricao = value;
	}

	/**
	 * Tipo de pergunta. verificar o enum TiposPergunta
	 * 
	 * @type {String}
	 * @memberOf Perguntaalinhamento
	 */
	public get tipo(): String {
		return this._tipo;
	}

	/**
	 * Tipo de pergunta. verificar o enum TiposPergunta
	 * 
	 * 
	 * @memberOf Pergunta
	 */
	public set tipo(value: String) {
		this._tipo = value;
	}

	/**
		 * Informa se a pergunta depende de outra pergunta para ser exibida.
		 * 
		 * @type {boolean}
		 * @memberOf Pergunta
		 */
	public get possuiDependencia(): boolean {
		return this._possuiDependencia;
	}

	/**
		 * Informa se a pergunta depende de outra pergunta para ser exibida.
		 * 
		 * @memberOf Pergunta
		 */
	public set possuiDependencia(value: boolean) {
		this._possuiDependencia = value;
	}

	/**
		 * Número de caracteres para os tipos de pergunta (TEXT e TEXTAREA)
		 * 
		 * @type {number}
		 * @memberOf Pergunta
		 */
	public get tamanho(): number {
		return this._tamanho;
	}

	/**
		 * Número de caracteres para os tipos de pergunta (TEXT e TEXTAREA)
		 * 
		 * @memberOf Pergunta
		 */
	public set tamanho(value: number) {
		this._tamanho = value;
	}

	/**
		 * Valor da resposta da pergunta
		 * 
		 * @type {string}
		 * @memberOf Pergunta
		 */
	public get resposta(): string {
		return this._resposta;
	}

	/**
		 * Valor da resposta da pergunta
		 * 
		 * @memberOf Pergunta
		 */
	public set resposta(value: string) {
		this._resposta = value;
	}

	/**
		 * Lista de opções para os tipos (RADIOBUTTON, CHECKBUTTON e SELECT)
		 * 
		 * @type {OpcaoPergunta[]}
		 * @memberOf Pergunta
		 */
	public get opcoes(): OpcaoPergunta[] {
		return this._opcoes;
	}

	/**
		 * Lista de opções para os tipos (RADIOBUTTON, CHECKBUTTON e SELECT)
		 * 
		 * @memberOf Pergunta
		 */
	public set opcoes(value: OpcaoPergunta[]) {
		this._opcoes = value;
	}

	/**
		 * Lista de perguntas dependentes desta pergunta
		 * 
		 * @type {PerguntaDependente[]}
		 * @memberOf Pergunta
		 */
	public get dependentes(): PerguntaDependente[] {
		return this._dependentes;
	}

	/**
		 * Lista de perguntas dependentes desta pergunta
		 * 
		 * @memberOf Pergunta
		 */
	public set dependentes(value: PerguntaDependente[]) {
		this._dependentes = value;
	}


	/**
	 * Getter formatoNumerico
	 * @return {FormatoNumerico}
	 */
	public get formatoNumerico(): FormatoNumerico {
		return this._formatoNumerico;
	}
	/**
	 * Setter formatoNumerico
	 * @param {FormatoNumerico} value
	 */
	public set formatoNumerico(value: FormatoNumerico) {
		this._formatoNumerico = value;
	}


    /**
     * Getter secao
     * @return {number}
     */
	public get secao(): number {
		return this._secao;
	}

    /**
     * Setter secao
     * @param {number} value
     */
	public set secao(value: number) {
		this._secao = value;
	}


    /**
     * Getter comValidacao
     * @return {boolean}
     */
	public get comValidacao(): boolean {
		return this._comValidacao;
	}

    /**
     * Setter comValidacao
     * @param {boolean} value
     */
	public set comValidacao(value: boolean) {
		this._comValidacao = value;
	}


    /**
     * Getter comErro
     * @return {boolean}
     */
	public get comErro(): boolean {
		return this._comErro;
	}

    /**
     * Setter comErro
     * @param {boolean} value
     */
	public set comErro(value: boolean) {
		this._comErro = value;
	}

    /**
     * Getter alinhamento
     * @return {string}
     */
	public get alinhamento(): string {
		return this._alinhamento;
	}

    /**
     * Setter alinhamento
     * @param {string} value
     */
	public set alinhamento(value: string) {
		this._alinhamento = value;
	}


	public jsonToEntity(res: any): Pergunta {

		if (res.opcoes) {
			this.opcoes = [];
			res.opcoes.forEach(opcao => {
				this.opcoes.push(new OpcaoPergunta().jsonToEntity(opcao));
			})			
		}

		if (res.dependentes) {
			this.dependentes = [];
			res.dependentes.forEach(dependente => {
				this.dependentes.push( new PerguntaDependente().jsonToEntity(dependente));
			});
		}

		if (res.formatoNumerico) {
			this.formatoNumerico = new FormatoNumerico().jsonToEntity(res.formatoNumerico);
		}		

		this.marcador 			= ConvertUtil.parseJsonParaAtributos(res.marcador, new String());
		this.id 				= ConvertUtil.parseJsonParaAtributos(res.id, new String());
		this.descricao 			= ConvertUtil.parseJsonParaAtributos(res.descricao, new String());
		this.tipo				= ConvertUtil.parseJsonParaAtributos(res.tipo, new String());
		this.possuiDependencia	= ConvertUtil.parseJsonParaAtributos(res.possuiDependencia, new Boolean());
		this.tamanho			= ConvertUtil.parseJsonParaAtributos(res.tamanho, new Number());
		this.resposta			= ConvertUtil.parseJsonParaAtributos(res.resposta, new String());
		this.secao				= ConvertUtil.parseJsonParaAtributos(res.secao, new Number());
		this.comValidacao		= ConvertUtil.parseJsonParaAtributos(res.comValidacao, new Boolean());
		this.comErro			= ConvertUtil.parseJsonParaAtributos(res.comErro, new Boolean());
		this.alinhamento		= ConvertUtil.parseJsonParaAtributos(res.alinhamento, new String());
		
		return this;
	}

}